src/blas_like/level1/MaxAbs.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+   Copyright (c) 2009-2016, Jack Poulson
+   All rights reserved.
+
+   This file is part of Elemental and is under the BSD 2-Clause License,
+   which can be found in the LICENSE file in the root directory, or at
+   http://opensource.org/licenses/BSD-2-Clause
+*/
+#include <El-lite.hpp>
+#include <El/blas_like/level1.hpp>
+
+namespace El {
+
+    template<typename Real,typename>
+    Real MaxAbs( const Matrix<Real>& A )
+    {
+        DEBUG_CSE
+        const Int m = A.Height();
+        const Int n = A.Width();
+        const Real* ABuf = A.LockedBuffer();
+        const Int ALDim = A.LDim();
+
+        Real value = 0;
+        for( Int j=0; j<n; ++j )
+            for( Int i=0; i<m; ++i )
+                value = Max(std::abs(value),ABuf[i+j*ALDim]);
+        return value;
+    }
+
+    template<typename Real,typename>
+    Real MaxAbs( const AbstractDistMatrix<Real>& A )
+    {
+        DEBUG_CSE
+        DEBUG_ONLY(
+          if( !A.Grid().InGrid() )
+              LogicError("Viewing processes are not allowed");
+        )
+        Real value = 0;
+        if( A.Participating() )
+        {
+            // Store the index/value of the local pivot candidate
+            const Int mLocal = A.LocalHeight();
+            const Int nLocal = A.LocalWidth();
+            const Real* ABuf = A.LockedBuffer();
+            const Int ALDim = A.LDim();
+            for( Int jLoc=0; jLoc<nLocal; ++jLoc )
+                for( Int iLoc=0; iLoc<mLocal; ++iLoc )
+                    value = Max(std::abs(value),ABuf[iLoc+jLoc*ALDim]);
+
+            value = mpi::AllReduce( value, mpi::MAX, A.DistComm() );
+        }
+        mpi::Broadcast( value, A.Root(), A.CrossComm() );
+        return value;
+    }
+
```

Comment:
I would recommend following the format of the Max routine: https://github.com/elemental/Elemental/blob/05f8445214f54149332ea12a739b7c10f407ecaa/src/blas_like/level1/Max.cpp#L128

For example, template functions must be explicitly instantiated. Elemental has a style for handling this that defines a `PROTO` macro, sets some definitions to guide which types to instantiate, and then includes the file `El/macros/Instantiate.h` to handle the instantiations given the macro(s) and guidance definitions.